Using redislabs/rejson with query api (#2305)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-redis.md

{{< tabs "Self-Hosted" "Kubernetes" "Azure" "AWS" "GCP" "Redis Enterprise Cloud" "Alibaba Cloud" >}}

{{% codetab %}}

> The Redis instance that gets created automatically with `dapr init` cannot be used with query API. You can run `redislabs/rejson` docker image on a different port(than already installed `redis` is using) to work with query API. 

Reviewer: ```suggestion
If you are using a self-hosted deployment of Dapr v1.7 and up, a Redis instance without the JSON module is automatically created as a Docker container when you run `dapr init`.

Alternatively, you can create an instance of Redis by running the following command:
 ```bash
 docker run -p 6379:6379 --name redis --rm redis
 ```

The Redis container that gets created on `dapr init` or via the above command, cannot be used with state store query API alone. You can run `redislabs/rejson` docker image on a different port(than already installed `Redis` is using) to work with query API. 
```

Reviewer 2: I would remove the mention to v1.7. This applies to all versions of Dapr and just makes it look stale one day.

> `redislabs/rejson` has support only for amd64 arch.

Reviewer: ```suggestion
> Note: `redislabs/rejson` has support only for amd64 architecture.
```

{{< tabs "Self-Hosted" "Kubernetes" "Azure" "AWS" "GCP" "Redis Enterprise Cloud" "Alibaba Cloud" >}}

{{% codetab %}}
If you are using a self-hosted deployment of Dapr, a Redis instance without the JSON module is automatically created as a Docker container when you run `dapr init`.

Alternatively, you can create an instance of Redis by running the following command:
 ```bash
 docker run -p 6379:6379 --name redis --rm redis
 ```

>  The Redis container that gets created on dapr init or via the above command, cannot be used with state store query API alone. You can run redislabs/rejson docker image on a different port(than already installed Redis is using) to work with query API.

Reviewer 3: ```suggestion
 The Redis container that gets created on dapr init or via the above command, cannot be used with state store query API alone. You can run redislabs/rejson docker image on a different port(than the already installed Redis is using) to work with query API.
```

>  The Redis container that gets created on dapr init or via the above command, cannot be used with state store query API alone. You can run redislabs/rejson docker image on a different port(than the already installed Redis is using) to work with query API.

Reviewer 3: ```suggestion
 The Redis container that gets created on dapr init or via the above command, cannot be used with state store query API alone. You can run redislabs/rejson docker image on a different port(than the already installed Redis is using) to work with they query API.
```

{{< tabs "Self-Hosted" "Kubernetes" "Azure" "AWS" "GCP" "Redis Enterprise Cloud" "Alibaba Cloud" >}}

{{% codetab %}}

> The Redis instance that gets created automatically with `dapr init` cannot be used with query api.

Reviewer 2: ```suggestion
The Redis instance that gets created automatically with `dapr init` cannot be used with query API.
```

> Use following command to create an instance of redis which can be used with query api:

Reviewer 2: ```suggestion
Use following command to create an instance of redis which can be used with query API:
```

```bash

> docker run -p 6379:6379 --name redis --rm redislabs/rejson:2.0.6

Reviewer: ask them to use a different port. 
Add in the component YAML file also ... .

Reviewer 2: Can we make sure all the steps are there if someone wants to do this locally 
1) Kill the current redis instance created with dapr init
2) Start the new instance with a different port just in case
3) Provide the YAML file for the query for redis.

Reviewer: if a different port is used, the existing Redis instance can also run side by side.
+1 to providing a default YAML file to connect. Also add link to the query state api doc for the component file modification needed to add query index etc.

Author: An example of component file is already provided in following lines. check line#311

{{< /tabs >}}

> Next is to start a Dapr application. Refer to this [component configuration file](../../../../developing-applications/building-blocks/state-management/query-api-examples/components/redis/redis.yml), which contains query indexing schemas. Kindly modify the `redisHost` to reflect the port on which `redislabs/rejson` is running.

Reviewer: ```suggestion
Next is to start a Dapr application. Refer to this [component configuration file](../../../../developing-applications/building-blocks/state-management/query-api-examples/components/redis/redis.yml), which contains query indexing schemas. Make sure to modify the `redisHost` to reflect the local forwarding port which `redislabs/rejson` uses.
```
